### ISSUE TITLE
cloud-init bond interface

### ISSUE THREAD

**Issue by radriaanse** [rancher/os#381]
Is there a way to setup a bond interface with cloud-init?

I know how to do it on for example CoreOS, but because the way you specify network settings is different in Rancher OS, it isn't clear on how to accomplish this if even possible.

**Comment by radriaanse** [comment]
To clarify, on CoreOS you would do something like this:
 
    - name: dhcp.network
      content: |
        [Match]
        Name=bond0

        [Network]
        DHCP=true
    - name: m1.network
      content: |
        [Match]
        Name=eth0

        [Network]
        Bond=bond0
    - name: m2.network
      content: |
        [Match]
        Name=eth1

        [Network]
        Bond=bond0
    - name: b1.netdev
      content: |
        [NetDev]
        Name=bond0
        Kind=bond

**Comment by bcg62** [comment]
I'm also looking to do something similar if possible

**Comment by MhdAli** [comment]
+1

**Comment by Rucknar** [comment]
Just like to say this is a feature we'd like to make use of to give us network redundancy on our physical hosts. Be good to know if this is just a case of using the right config or if it requires a feature coding.

**Comment by joshwget** [comment]
I think this functionality should be supported with our recent networking changes (http://docs.rancher.com/os/configuration/networking/).

**Comment by imikushin** [comment]
This is totally possible In RancherOS. For example:
```yaml
#cloud-config
rancher:
  network:
    interfaces:
      bond0:
        addresses:
        - 192.168.101.33/31
        - 10.88.23.129/31
        gateway: 192.168.101.32
        bond_opts:
          downdelay: "200"
          lacp_rate: "1"
          miimon: "100"
          mode: "4"
          updelay: "200"
          xmit_hash_policy: layer3+4
        post_up:
        - ip route add 10.0.0.0/8 via 10.88.23.128
      mac=0c:c4:d7:b2:14:d2:
        bond: bond0
      mac=0c:c4:d7:b2:14:d3:
        bond: bond0
```

**Comment by imikushin** [comment]
rancher/rancher.github.io#343

**Comment by deniseschannon** [comment]
This will be released in the docs when v0.5.0 is released
